rmgpy/molecule/resonance.py

```diff
@@ -341,17 +389,187 @@ def generate_isomorphic_isomers(mol):
+        a.append(inRing + inBond)
+
+    # Objective vector for optimization: sextets have a weight of 1, double bonds have a weight of 0
+    objective = [1] * l + [0] * len(bonds)
+
+    # Solve LP problem using lpsolve
+    lp = lpsolve('make_lp', m, n)               # initialize lp with constraint matrix with m rows and n columns
+    lpsolve('set_verbose', lp, 2)               # reduce messages from lpsolve
+    lpsolve('set_obj_fn', lp, objective)        # set objective function
+    lpsolve('set_maxim', lp)                    # set solver to maximize objective
+    lpsolve('set_mat', lp, a)                   # set left hand side to constraint matrix
+    lpsolve('set_rh_vec', lp, [1] * m)          # set right hand side to 1 for all constraints
+    lpsolve('set_constr_type', lp, ['='] * m)   # set all constraints as equality constraints
+    lpsolve('set_binary', lp, [True] * n)       # set all variables to be binary
+
+    # Constrain values of exocyclic bonds, since we don't want to modify them
+    for i in range(l, n):
+        if exo[i - l] is not None:
+            # NOTE: lpsolve indexes from 1, so the variable we're changing should be i + 1
+            lpsolve('set_bounds', lp, i + 1, exo[i - l], exo[i - l])
+
+    # Add constraints to problem if provided
+    if constraints is not None:
+        for constraint in constraints:
+            lpsolve('add_constraint', lp, constraint[0], '<=', constraint[1])
+
+    status = lpsolve('solve', lp)
+    objVal, solution = lpsolve('get_solution', lp)[0:2]
+    lpsolve('delete_lp', lp)  # Delete the LP problem to clear up memory
+
+    # Check that optimization was successful
+    if status != 0:
+        raise RuntimeError('Optimization did not obtain optimal solution')
+
+    # Check that we the result contains at least one aromatic sextet
+    if objVal == 0:
+        return []
+
+    # Check that the solution contains the maximum number of sextets possible
+    if maxNum is None:
+        maxNum = objVal  # This is the first solution, so the result should be an upper limit
+    elif objVal < maxNum:
+        raise ValueError('Sub-optimal solution obtained.')
+
+    if any([x != 1 and x != 0 for x in solution]):
+        raise ValueError('Non-integer solution obtained from optimization.')
+
+    # Generate constraints based on the solution obtained
+    y = solution[0:l]
+    new_a = y + [0] * len(bonds)
+    new_b = sum(y) - 1
+    if constraints is not None:
+        constraints.append((new_a, new_b))
+    else:
+        constraints = [(new_a, new_b)]
+
+    # Run optimization with additional constraints
+    try:
+        innerSolutions = clarOptimization(mol, constraints=constraints, maxNum=maxNum)
+    except (ValueError, RuntimeError):
+        innerSolutions = []
+
+    return innerSolutions + [(molecule, ASSSR, bonds, solution)]
+
+
+def clarTransformation(mol, ring):
+    """
+    Performs Clar transformation for given ring in a molecule, ie. conversion to aromatic sextet.
```

Comment:
made it as `def clarTransformation(mol, aromaticRing)` to imply it's only for aromatic ring